src/credentials/sharedCredentialsValidation.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * This module focuses on the validation of shared credentials properties
+ */
+
+import { localize } from 'vscode-nls'
+import { SharedCredentialsKeys, StaticCredentialsProfileData } from './types'
+
+/**
+ * The format validators for shared credentials keys.
+ *
+ * A format validator validates the format of the data,
+ * but not the validity of the content.
+ */
+export const CredentialsKeyFormatValidators = {
+    [SharedCredentialsKeys.AWS_ACCESS_KEY_ID]: getAccessKeyIdFormatError,
+    [SharedCredentialsKeys.AWS_SECRET_ACCESS_KEY]: getSecretAccessKeyFormatError,
+} as const
+
+/**
+ * Holds the error for each key of static credentials data,
+ * if it exists. This allows the user to get all the errors
+ * at once.
+ */
+export type StaticCredentialsErrorResult = {
+    [k in keyof StaticCredentialsProfileData]: string | undefined
+}
+
+export function getStaticCredentialsDataErrors(
+    data: StaticCredentialsProfileData
+): StaticCredentialsErrorResult | undefined {
```

Comment:
I see this function as a convenience function which picks the correct functions from `CredentialsKeyFormatValidators`. If we find this doesn't work it will be easy to change

Reply:
The validators are 1:1 with the keys though, would that ever not be true? I'm being nitpicky though so I'd keep the current impl.